src/gui/properties/peerlistwidget.cpp

```diff
@@ -179,15 +182,17 @@ void PeerListWidget::showPeerListMenu(const QPoint&)
   QAction *act = menu.exec(QCursor::pos());
   if (act == 0) return;
   if (act == addPeerAct) {
-    BitTorrent::PeerAddress addr = PeerAdditionDlg::askForPeerAddress();
-    if (!addr.ip.isNull()) {
-      if (torrent->connectPeer(addr))
-        QMessageBox::information(0, tr("Peer addition"), tr("The peer was added to this torrent."));
-      else
-        QMessageBox::critical(0, tr("Peer addition"), tr("The peer could not be added to this torrent."));
-    }
-    else {
-      qDebug("No peer was added");
+    QList<BitTorrent::PeerAddress> peersList = PeersAdditionDlg::askForPeersEndpoints();
+    if (!peersList.isEmpty()) {
```

Comment:
1. Leave it as is.
2. Don't show message box at all.
3. To track whether peers were really added.